src/test/java/htsjdk/samtools/SamFlagFieldTest.java

```diff
@@ -0,0 +1,140 @@
+package htsjdk.samtools;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * @author nhomer
+ */
+public class SamFlagFieldTest {
+
+    @Test
+    public void testAllFlags() {
+        int flagAsInteger = 0;
+        for (final SAMFlag samFlag : SAMFlag.values()) {
+            flagAsInteger |= samFlag.flag;
+        }
+        final String flagAsString = SamFlagField.STRING.format(flagAsInteger);
+
+        Assert.assertEquals(flagAsString, "urURpP12sSxd");
+    }
+
+    @Test
+    public void testAllFlagsReverseOrder() {
+        int flagAsInteger = 0;
+        for (final SAMFlag samFlag : SAMFlag.values()) {
+            flagAsInteger |= samFlag.flag;
+        }
+        final String flagAsString = new StringBuilder("urURpP12sSqd").reverse().toString();
```

Comment:
I need to learn this gradle business if I am to test locally.  Off to learn gradle.